.github/GITHUB_WORKFLOW.md

```diff
@@ -0,0 +1,301 @@
+
+Before merging a PR, squash the following kinds of commits:
+
+- Fixes/review feedback
+- Typos
+- Merges and rebases
+- Work in progress
+
+Aim to have every commit in a PR compile and pass tests independently if you can, but it's not a requirement.
+
+To squash your commits, perform an [interactive rebase](https://git-scm.com/book/en/v2/Git-Tools-Rewriting-History):
+
+1. Check your git branch:
+
+  ```bash
+  git status
+  ```
+
+  The output should be similar to this:
+
+  ```bash
+  On branch your-contribution
+  Your branch is up to date with 'origin/your-contribution'.
+  ```
+
+1. Start an interactive rebase using a specific commit hash, or count backwards from your last commit using `HEAD~<n>`, where `<n>` represents the number of commits to include in the rebase.
+
+  To get this `<n>` value, the easiest way to doing this is running `git log` and then count number of commits until you hit `main` (this may be called `master` instead depending on the repo) but not including `main`.
+
+  ```bash
+  git log
+  ```
+
+  Once you have that number, continue with the rebase and squash.
+
+  ```bash
+  git rebase -i HEAD~3
+  ```
+
+  The output should be similar to this:
+
+  ```bash
+  pick 2ebe926 Original commit
+  pick 31f33e9 Address feedback
+  pick b0315fe Second unit of work
+
+  # Rebase 7c34fc9..b0315ff onto 7c34fc9 (3 commands)
+  #
+  # Commands:
+  # p, pick <commit> = use commit
+  # r, reword <commit> = use commit, but edit the commit message
+  # e, edit <commit> = use commit, but stop for amending
+  # s, squash <commit> = use commit, but meld into previous commit
+  # f, fixup <commit> = like "squash", but discard this commit's log message
+  ...
+  ```
+
+1. Use a command line text editor to change the word `pick` to `squash` for the commits you want to squash, then save your changes and continue the rebase:
+
+  ```bash
+  pick 2ebe926 Original commit
+  squash 31f33e9 Address feedback
+  squash b0315fe Second unit of work
+  ...
+  ```
+
+  The output after saving changes should look similar to this:
+
+  ```bash
+  [detached HEAD 61fdded] Second unit of work
+   Date: Thu Mar 5 19:01:32 2020 +0100
+   2 files changed, 15 insertions(+), 1 deletion(-)
+   ...
+
+  Successfully rebased and updated refs/heads/main.
+  ```
+
+1. Force push your changes to your remote branch:
+
+  ```bash
+  git push --force
+  ```
+
+By squashing locally, you control the commit message(s) for your work, and can separate a large PR into logically separate changes. For example: you have a pull request that is code complete and has 24 commits. You rebase this against the same merge base, simplifying the change to two commits. Each of those two commits represents a single logical change and each commit message summarizes what changes. Reviewers see that the set of changes are now understandable, and approve your PR.
+
+## 11. Merging a commit
```

Comment:
Please can we use rebase. If we're setting up with squashes, we should be prepped for a rebase merge at this point.